api/v1alpha2/machinepool_types.go

```diff
@@ -0,0 +1,164 @@
+/*
```

Comment:
@vincepri @detiber controller-gen was giving me problems if the type didn't already exist in v1alpha2. Any idea what to do here?

Reply:
What was the error message? We should definitely not add the type to v1alpha2

Reply:
Try adding this to the v1a3 type: `+k8s:conversion-gen=false`?